File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    int[] one = {1,2,3,4,5};
    int[] two = {2,1,2,3,2,4,2,5};
    int[] three = {3,3,1,1,2,2,4,4,5,5};
    
    public int[] solution(int[] answers) {
        int[] score = new int[3];
        for(int i = 0; i< answers.length; i++){
            if(one[i%5] == answers[i]){
                score[0]++;
            }
            if(two[i%8] == answers[i]){
                score[1]++;
            }
            if(three[i%10] == answers[i]){
                score[2]++;
            }
        }
        
        int max = 0;
        for(int n : score){
            max = Math.max(n, max);
        }
        Queue<Integer> que = new ArrayDeque();
        for(int i=0; i<3; i++){
            if(score[i] == max){
                que.offer(i+1);
            }
        }
        int[] answer = new int[que.size()];
        int idx = 0;
        while(!que.isEmpty()){
            answer[idx] = que.poll();
            idx++;
        }
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    a = [1,2,3,4,5] * 2000
    b = [2,1,2,3,2,4,2,5] * 1250
    c = [3,3,1,1,2,2,4,4,5,5] * 1000
    answer = []
    score = [0,0,0]
    
    for idx, val in enumerate(answers):
        if a[idx] == val:
            score[0] += 1
        if b[idx] == val:
            score[1] += 1
        if c[idx] == val:
            score[2] += 1
    
    max_score = max(score)
    
    for i,v in enumerate(score):
        if max_score == v:
            answer.append(i+1)
    
    return answer